### PR TITLE
Repair profile picture upload

### PR DESCRIPTION
**Pull request by eliandoran**
The following issues were fixed:

1. `FormData` is an HTML5 construct, thus requiring a DOM Object, not a jQuery selector result.
2. The AJAX back-end tried to access the profile picture POST field using different names, which caused errors.

**Please do not forget to create the `profile_pictures/` directory in the same directory as `ajax-signup.php`.**

### CODE REVIEW

File: register.js [hunk 1]
```diff
@@ -151,7 +151,7 @@ $('document').ready(function()
 				$.ajax({
 			   		url: 'ajax-signup.php',
 			   		type: 'POST',
-					data: new FormData($('#register-form')),
+					data: new FormData(document.getElementById("register-form")),
```
Comment (eliandoran) [inline register.js]: You were trying to pass a jQuery selector result (which is some kind of iterator function) to a HTML5 `FormData` constructor which wanted a `<form>` DOMElement. 

_`$('#register-form')` is not equal to `document.getElementById("register-form")`_.

See <https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData> for `FormData` constructor documentation.

File: ajax-signup.php [hunk 2]
```diff
@@ -17,22 +17,20 @@
...
 		/* Storing the profile picture in disk and doing validation */
         /* -- Extracting file extensions and filename for renaming the file later on. */
         $file_property = pathinfo($user_picture);
         $pic_filename = $file_property['filename'];
         $pic_file_extension = $file_property['extension'];
-        $directory = "/var/www/html/open-learning/profile_pictures/";
+        $directory = dirname(__FILE__) . "/profile_pictures/";
 		$directory .= $pic_filename.rand().'.'.$pic_file_extension;
-
-		move_uploaded_file($_FILES['picture']['tmp_name'], $directory); //Moving file to the disk
```
Comment (eliandoran) [inline ajax-signup.php]: You used `picture` to refer to the uploaded file, but you previously referred to it (in the variable declaration) as `profile_picture`.

The name of the form field needs to match the one in the HTML, which happened to be actually `profile_picture`. You were trying to read a non-existing field.

File: ajax-signup.php [hunk 3]
```diff
@@ -17,22 +17,20 @@
...
 		//Password hashing
 		$hashed_password = password_hash($user_pass, PASSWORD_DEFAULT);
 
-
 		/* Storing the profile picture in disk and doing validation */
         /* -- Extracting file extensions and filename for renaming the file later on. */
         $file_property = pathinfo($user_picture);
         $pic_filename = $file_property['filename'];
         $pic_file_extension = $file_property['extension'];
-        $directory = "/var/www/html/open-learning/profile_pictures/";
```
Comment (eliandoran) [inline ajax-signup.php]: Hard-coded paths are evil, @SDey96 .

File: ajax-signup.php [hunk 4]
```diff
@@ -17,22 +17,20 @@
...
+        $directory = dirname(__FILE__) . "/profile_pictures/";
```
Comment (eliandoran) [inline ajax-signup.php]: Much better, make the path relative to the current file.

File: ajax-signup.php [hunk 5]
```diff
@@ -17,22 +17,20 @@
 		$user_pass = strip_tags($password);
 		$website = $_POST['website'];
 		$twitter = $_POST['twitter'];
-		//$user_picture = $_FILES['profile_picture']['name'];
+		$user_picture = $_FILES['profile_picture']['name'];
```
Comment (eliandoran) [inline ajax-signup.php]: This is actually not even needed.

File: ajax-signup.php [hunk 6]
```diff
@@ -66,7 +63,7 @@
         } else {
             $response['status'] = 'error'; // could not register
 			$response['message'] = '<span class="glyphicon glyphicon-info-sign"></span> &nbsp; Could not register now. Please try again later.';
-			echo $_FILES['profile_pic'];
+			$response["query"] = mysqli_error($link);
```
Comment (eliandoran) [inline ajax-signup.php]: For debugging purposes, I also outputted MySQL errors to the response, which I could track when doing POST request using Postman.

**Remove in production, @SDey96**

**Comment by SDey96** [comment]
Thank you very much, Elian!

File: register.js (hunk 1 above)
Comment (SDey96) [reply register.js]: Agreed, this is what was causing the issue. And searching over Google helped me find out that "serialize()" function in jQuery does not support uploading files through AJAX. Moreover, the original file could be found in the "register(copy).js". The file, "register.js" had my altered modifications, which added more bugs.

File: ajax-signup.php (hunk 4 above)
Comment (SDey96) [reply ajax-signup.php]: Alternately, you can do:
``$directory = getcwd();``

File: ajax-signup.php (hunk 3 above)
Comment (SDey96) [reply ajax-signup.php]: I know, but when nothing works, you hard code things. Earlier I used ``$directory=getcwd()`` instead. But, when the bad luck kept following me, I hard coded it.

File: ajax-signup.php (hunk 4 above)
Comment (eliandoran) [reply ajax-signup.php]: Yes, although there's a slight difference when the PHP file is included.

`dirname(__FILE__)` is relative to the path being included.
`getcwd()` is relative to the path of the script actually being executed.

See <https://stackoverflow.com/questions/2184810/difference-between-getcwd-and-dirname-file-which-should-i-use> for further details.

**Comment by eliandoran** [comment]
You're welcome, @SDey96 .

File: ajax-signup.php (hunk 6 above)
Comment (SDey96) [reply ajax-signup.php]: Thanks, I removed it already.

File: ajax-signup.php (hunk 4 above)
Comment (SDey96) [reply ajax-signup.php]: Thank you, I will consider that next time.

File: ajax-signup.php (hunk 2 above)
Comment (SDey96) [reply ajax-signup.php]: I know how it works. Previously, I called the form attribute ``picture`` and matched all the references in the aforementioned PHP file. But, when I failed to upload the file to the disk, I then renamed that attribute to ``profile_picture`` and its every reference, too. I might have skipped to fix that line later.

**Pull request merged**